Investigate static analysis via lgtm.com

We should check whether we can perform static analysis with lgtm.com. I added [Spicy](https://lgtm.com/projects/g/zeek/spicy/?mode=list), but this fails without a custom config.

I tried the following custom config:

```.yml
path_classifiers:
  library:
    - "3rdparty"

extraction:
  cpp:
    prepare:
      packages:
        - git
        - ninja-build
        - bison
        - flex
        - bison
        - llvm-9-dev
        - clang-9
        - libclang-9-dev
        - libc++1
        - libc++abi1
        - libc++abi-dev
        - curl
    after_prepare:
        - cd /usr/local/cmake && curl -L https://github.com/Kitware/CMake/releases/download/v3.15.0/cmake-3.15.0-Linux-x86_64.tar.gz | tar xzvf - -C /usr/local/cmake --strip-components 1
        - export PATH=/usr/local/cmake/bin:$PATH
        - export CXX=clang++-9
        - export CC=clang-9
        - export ASM=clang-9

    configure:
      command:
        - ./configure --generator=Ninja

    index:
      build_command:
        - ninja -C build
```

This installs a custom CMake, since with upstream package in lgtm's setup is too old. However, it seems that lgtm injects a custom cmake wrapper in order to capture build commands which is also too old (3.13.4). With that I believe we are blocked by lgtm bumping their CMake version, or by finding another way to produce the artifacts it expects.

```
[2020-12-16 15:22:33] [build] CMake Error at CMakeLists.txt:3 (cmake_minimum_required):
[2020-12-16 15:22:33] [build]   CMake 3.15.0 or higher is required.  You are running version 3.13.4
[2020-12-16 15:22:33] [build] -- Configuring incomplete, errors occurred!
```

--- a/.lgtm.yml
+++ b/.lgtm.yml
@@ -1,0 +1,36 @@
+path_classifiers:
+  library:
+    - "3rdparty"
+
+extraction:
+  cpp:
+    prepare:
+      packages:
+        - git
+        - ninja-build
+        - bison
+        - flex
+        - bison
+        - llvm-9-dev
+        - clang-9
+        - libclang-9-dev
+        - libc++1
+        - libc++abi1
+        - libc++abi-dev
+        - curl
+    after_prepare:
+        - mkdir $HOME/cmake && curl -L https://github.com/Kitware/CMake/releases/download/v3.15.0/cmake-3.15.0-Linux-x86_64.tar.gz | tar xzvf - -C $HOME/cmake --strip-components 1
+        - export PATH=$HOME/cmake/bin:$PATH
+        - export CXX=clang++-9
+        - export CC=clang-9
+        - export ASM=clang-9
+        - git fetch --tags
+
+    configure:
+      command:
+        - cmake --version
+        - ./configure --generator=Ninja
+
+    index:
+      build_command:
+        - ninja -C build

